File: src/485_MaxConsecutiveOnes/485.go
func Max(a int, b int) int {
    if a > b { return a }
    return b 
}

func findMaxConsecutiveOnes(nums []int) int {
    max := 0 
    curr := 0
    for _, n := range nums {
        fmt.Println(n, curr)
        if n == 0 {
            max = Max(max, curr)
            curr = 0
        } else {
            curr += 1
        }
    }

    return Max(max, curr)
}

File: src/485_MaxConsecutiveOnes/485.rb
# @param {Integer[]} nums
# @return {Integer}
def find_max_consecutive_ones(nums)
  def groups(nums, carry=[[]])
    return carry if nums == [] 
    head = nums.pop
    if head == 1 
        carry[-1] << head
    else 
        carry << []
    end
    return groups(nums, carry)
  end

  return groups(nums).max_by {|g| g.size }.size 
end
